### ISSUE TITLE
kubelet makes a lot of noise complaining about missing volumes subdir for static pods

### ISSUE THREAD

**Issue by errordeveloper** [kubernetes/kubernetes#31863]
I keep seeing

```
E0831 17:11:32.468072    2136 kubelet_getters.go:248] Could not read directory /var/lib/kubelet/pods/b608eb2905506727f838b3c31f596013/volumes: open /var/lib/kubelet/pods/b608eb2905506727f838b3c31f596013/volumes: no such file or directory
```

in kubelet logs on master, and that directory is actually missing indeed.

This pod is one of the static pods for scheduler, and this one doesn't have any volumes.

The fix to this may be related to https://github.com/kubernetes/kubernetes/issues/31272.

cc @saad-ali @jingxu97 @pmorie

**Comment by jingxu97** [comment]
@errordeveloper The reason for this error message is because there is no volume directory under /var/lib/kubelet/pods/ UID/. But the code in here https://github.com/kubernetes/kubernetes/blob/master/pkg/kubelet/kubelet_volumes.go#L150 assumes there is always a volume directory under pod dir. Will work on a fix for this.

**Comment by errordeveloper** [comment]
SGTM

**Comment by pmorie** [comment]
@errordeveloper In this case, is `b608eb2905506727f838b3c31f596013` the UID of the static pod or the mirror pod?

**Comment by goltermann** [comment]
Moving out of v1.4 as this is a P2.

**Comment by errordeveloper** [comment]
Good question, Paul, but I don't know. I will need reproduce.

**Comment by dims** [comment]
switching this to 1.6 as it's too late for 1.5. ok? (please switch it right back if you disagree)

**Comment by dchen1107** [comment]
@jingxu97 I believe you already had the patch fixing this one a while back. If I am right, please close this one.

**Comment by jingxu97** [comment]
Thanks, @dchen1107. Yes, the fix is #38909. I think we can close it now.
